cli/module_registry.go

```diff
@@ -176,46 +174,152 @@ func UpdateModuleCommand(c *cli.Context) error {
-				manifestPath, org.Name, org.PublicNamespace)
+				defaultManifestFilename, org.Name, org.PublicNamespace)
 			fmt.Fprintf(c.App.Writer, "You no longer need to specify org_id or public_namespace\n")
 		}
 	}
 
 	return nil
 }
 
+// UploadModuleCommand runs the command to upload a new version of a module.
+func UploadModuleCommand(c *cli.Context) error {
+	manifestPathArg := c.String("module")
+	publicNamespaceArg := c.String("public_namespace")
+	orgIDArg := c.String("org_id")
+	nameArg := c.String("name")
+	versionArg := c.String("version")
+	platformArg := c.String("platform")
+	tarballPath := c.Args().First()
+	if c.Args().Len() > 1 {
+		return errors.New("Too many arguments passed to upload command. " +
+			"Make sure to specify flag and optional arguments before the required positional package argument")
+	}
+	if tarballPath == "" {
+		return errors.New("No package to upload -- please provide an archive containing your module. See the help for more information")
+	}
+
+	client, err := NewAppClient(c)
+	if err != nil {
+		return err
+	}
+
+	manifestPath := defaultManifestFilename
+	if manifestPathArg != "" {
+		manifestPath = manifestPathArg
+	}
+	var moduleID ModuleID
+	// if the manifest cant be found
+	if _, err := os.Stat(manifestPath); err != nil {
+		// no manifest found.
+		if nameArg == "" || (publicNamespaceArg == "" && orgIDArg == "") {
+			return errors.Errorf("Unable to find %s. "+
+				"If you want to upload a version without a %s, you must supply a module name and namespace (or module name and orgid)\n\n",
+				defaultManifestFilename, defaultManifestFilename)
+		}
+		moduleID = ModuleID{Prefix: publicNamespaceArg, Name: nameArg}
+	} else {
+		// if we can find a manifest, use that
+		manifest, err := loadManifest(manifestPath)
+		if err != nil {
+			return err
+		}
+
+		moduleID, err = parseModuleID(manifest.Name)
+		if err != nil {
+			return err
+		}
+		if nameArg != "" && nameArg != moduleID.Name {
+			// This is almost certainly a mistake we want to catch
+			return errors.Errorf("Module name %q was supplied via command line args but the %s has a module name of %q",
+				nameArg, defaultManifestFilename, moduleID.Name)
+		}
+	}
+	// TODO(zaporter) This logic is duplicated in update and upload and should be refactored
+	// TODO(APP-2230) Simplify this once we can pass org-id in ModuleID
+	if publicNamespaceArg != "" {
+		switch moduleID.Prefix {
+		case "":
+			moduleID.Prefix = publicNamespaceArg
+		case publicNamespaceArg:
+			// the meta.json manifest == public_namespace arg
+			fmt.Fprintf(c.App.Writer, "The module's %s already specifies a public namespace. Ignoring\n", defaultManifestFilename)
+		default:
+			// the meta.json manifest != public_namespace arg
+			// we may want to investigate a better way of handling this error case
+			// For now, it seems like bad UX to ignore this error
+			return errors.Errorf("the module's %s specifies a namespace of '%s'"+
```

Comment:
instead of doing + you can do `` to wrap around your error string across multi-lines. Looks a lot cleaner

Reply:
But this isn't for multi-line strings, this is to get around the 140 char line limit for our linter. Am I misunderstanding your comment?

Reply:
You need to make it multi-line so that it doesn't break the 140 char limit. So you can do that using `` rather than + to not break the limit.